Admin/AdminBuilderInterface.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Admin;
+
+use Sonata\AdminBundle\Datagrid\DatagridInterface;
+use Symfony\Component\Form\FormInterface;
+
+/**
+ * Builds admin. There is a dependency on the AdminInterface because
+ * this object holds useful object to deal with this task, but there is
+ * probably a better design.
+ *
+ * @author Christian Gripp <mail@core23.de>
+ */
+interface AdminBuilderInterface
```

Comment:
Might be misinterpreted as something that builds admin objects… instead, it builds forms, and datagrids. Hard to name it better b/c it builds a datagrid, otherwise it would simply be `AdminFormBuilderInterface`. Maybe you should move the last method to a separate interface, and have `AdminBuilder` implement both interfaces? Or even go one step further and also have two classes?